### ISSUE TITLE
Cross-linking

### ISSUE THREAD

**Issue by opoudjis** [metanorma/metanorma.com#115]
From #109:

More to the point, there needs to be generous crosslinking between pages, so that if you find something similar in the menu, the similar page can steer you to the page you actually need.
https://www.metanorma.com/author/topics/document-format/bibliography/#automatic-reference-fetching for example needs to cross-reference https://www.metanorma.com/author/topics/building/reference-lookup/ .

I think there's a reasonable amount of cross-linking already, but it's worth someone (not me right now) doing it as a project.

Specifically for flavours, we should be linking from the specific back to the corresponding generic documentation throughout, for every section in flavour documentation that corresponds to generic documentation. (The alternative is clever use of includes, so that the flavour-specific documentation can stand on its own. We did intend to do that, but it's difficult, and if we have good cross-linking, it may not be necessary.)

So for example all listings of document attributes need to link back to the generic standoc listing of document attributes. All discussion of section markup needs to link back to the generic standoc documenation of section markup. And so on.

**Comment by opoudjis** [comment]
I'll do some of this as time permits, but I do think this needs to be a systematic project.

**Comment by strogonoff** [comment]
Yeah, this kind of cross-linking is tough to implement automatically.

**Comment by strogonoff** [comment]
Here is the suggested algorithm to start with:

1. Pick a topic in general Metanorma docs that has its own page.

2. Go through each flavor, and see if there is a section in flavor docs that corresponds to that topic.

3. If such a section exists, add a “Note” block in the beginning of it, linking to aforementioned topic page in general docs.

cc @opoudjis

**Comment by strogonoff** [comment]
This implies a lot of duplication, since each Note block would have similar copy inside, only link and topic name would change.

To reduce the duplication, I will add a sort of template, so @anermina will be able to insert a snippet and only specify a link and topic title (like `{% include general-crosslink.html url=… topic_title=… %}`). However, no need to wait for that before beginning with the first Note block.

**Comment by strogonoff** [comment]
Example of note block syntax can be found [here](https://github.com/metanorma/metanorma.com/blob/master/author/topics/document-format/diagrams.adoc). If you see the raw source, it looks as follows:

```
[NOTE]
====
This requires PlantUML to be installed and available. If you installed the `metanorma-cli` gem manually,
make sure to read link:/software/metanorma-cli/docs/installation/[installation notes].
====
```

**Comment by opoudjis** [comment]
@strogonoff Can you please take this ticket over and review @anermina's work?

**Comment by strogonoff** [comment]
> @strogonoff Can you please take this ticket over and review @anermina's work?

Sure, we’re sorting out pull request initiation as of now

**Comment by strogonoff** [comment]
Nope, my bad, PR is created, only GH didn’ show it here

**Comment by strogonoff** [comment]
For the record, here is what the first note looks like:

<img width="1162" alt="Screen Shot 2019-10-09 at 7 38 22 PM" src="https://user-images.githubusercontent.com/292959/66478186-6463c980-eacc-11e9-9c47-a0870b9bae28.png">

I asked @anermina to add notes in the beginning of sections, after header.
